lib/sinatra/main.rb

```diff
@@ -1,16 +1,8 @@
 require 'sinatra/base'
 
 module Sinatra
-  class Application < Base
-
-    # we assume that the first file that requires 'sinatra' is the
-    # app_file. all other path related options are calculated based
-    # on this path by default.
-    set :app_file, caller_files.first || $0
-
-    set :run, Proc.new { File.expand_path($0) == File.expand_path(app_file) }
-
-    if run? && ARGV.any?
```

Comment:
Anyone have thoughts on why `run?` needed to be true for this condition?

Reply:
Probably no problem.

Reply:
I reconsidered this, and we need to check by `run?` coz we need to take care that unexpected `set`s are not called. I'll fix it soon.